debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/UnwrapFromMongoDbEnvelope.java

```diff
@@ -166,6 +162,17 @@ public R apply(R r) {
                     }
                 }
 
+                if (!document.containsKey("$set") && !document.containsKey("$unset")) {
```

Comment:
Very nice! Also great with the separate test commits. 👍 It really makes it easy to review.

Reply:
I'm glad you like it! :)